kong/core/router.lua

```diff
@@ -464,7 +464,7 @@ function _M.new(apis)
     method = upper(method)
 
 
-    local host = headers["host"] or headers["Host"]
+    local host = headers["X-Host-Override"] or headers["x-host-override"] or headers["host"] or headers["Host"]
```

Comment:
It was originally when writing [those](https://github.com/Mashape/kong/blob/master/spec/01-unit/11-router_spec.lua) tests iirc. None appear to use the capitalized form anymore, but I'd rather keep it just in case, to avoid future confusion.